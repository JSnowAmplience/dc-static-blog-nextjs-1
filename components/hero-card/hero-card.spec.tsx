/* eslint-env jest */
import renderer from 'react-test-renderer';
import HeroCard from './hero-card';
import blogPostFixture from '../../tests/fixtures/single-blog-post-data-object.json';

const mockUseRouter = jest.fn();
jest.mock('next/router', () => {
  return {
    useRouter: () => mockUseRouter()
  };
});

describe('HeroCard', () => {
  test('renders full hero card', async () => {
    mockUseRouter.mockImplementationOnce(() => {
      return { query: {} };
    });
    const component = await renderer.create(<HeroCard blogPost={blogPostFixture} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders full hero card with vse query string in link for preview', async () => {
    mockUseRouter.mockImplementationOnce(() => {
      return { query: { vse: 'test-vse.domain' } };
    });
    const component = await renderer.create(<HeroCard blogPost={blogPostFixture} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders empty hero card when blog list is undefined', async () => {
    const props = {
      blogPosts: undefined
    };
    //@ts-ignore
    const component = await renderer.create(<HeroCard {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
