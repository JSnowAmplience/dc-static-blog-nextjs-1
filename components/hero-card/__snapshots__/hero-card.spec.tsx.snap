// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCard renders empty hero card when blog list is undefined 1`] = `<div />`;

exports[`HeroCard renders full hero card 1`] = `
<section
  className="jsx-238292447"
>
  <a
    aria-label="blog-post-test"
    className="jsx-580416171"
    href="/blog/url-slug/undefined"
  >
    <article
      className="jsx-238292447"
    >
      <div
        className="jsx-238292447 blog-card-image"
      >
        <div
          className="image-mock"
        />
      </div>
      <div
        className="jsx-238292447 blog-card-content"
      >
        <h1
          className="jsx-238292447"
        >
          blog-post-test
        </h1>
        <div
          className="blog-card-meta-mock"
        />
        <p
          className="jsx-238292447"
        >
          blog-post-description
        </p>
      </div>
    </article>
  </a>
</section>
`;
